app/models/url.rb

```diff
@@ -0,0 +1,21 @@
+require "uri"
+class Url < ActiveRecord::Base
+  validates :long, :uniqueness => true, format: { with: /\A#{URI::regexp}\z/}
+  before_save :convert
+
+  private
+
+  def convert
+     range = [*'0'..'9', *'a'..'z', *'A'..'Z']
+     unless self.short
+        short_url = Array.new(8){range.sample}.join
```

Comment:
If you're feeling really clever, you could do this:

    len = 3
    until self.short
      short_url = Array.new(len){range.sample}.join
      self.short = short_url unless Url.find_by(short_url: short_url)
      len = len + 1
    end